hooks/useBotsNames.ts

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useState } from 'react';
+import { resolveUserID } from '../utils';
+import { useAppDispatch, useAppSelector } from '../redux/hooks';
+import { Alert, LocationCredential } from '../types';
+import { addAlert } from '../redux/alertsSlice';
+import { setStartRetrievingBotsNames } from '../redux/adminSlice';
+
+export const useBotsNames = () => {
+  const dispatch = useAppDispatch();
+
+  const { isWaitingForBotsNames, botsNames } = useAppSelector(
+    state => state.admin,
+  );
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [botsAvailable, setBotsAvailable] = useState(false);
+  const [locationCredentials, setLocationCredentials] = useState<
+    LocationCredential[]
+  >([]);
+
+  useEffect(() => {
+    if (isWaitingForBotsNames) {
+      return;
+    }
+    if (!!botsNames.length) {
```

Comment:
```suggestion
    if (botsNames.length) {
```
redundant double negation